Services/housekeeping_service.c

```diff
@@ -55,7 +55,26 @@ SAT_returnState hk_service_app(csp_packet_t *packet) {
 }
 
 static SAT_returnState hk_parameter_report(csp_packet_t *packet) {
-  HAL_hk_report(packet->data[SID_byte], packet->data + SID_byte + 1);
+  if(HAL_hk_report(packet->data[SID_byte], packet->data + SID_byte + 1) != SATR_OK) {
+    ex2_log("Failed to collecting report data, err src: %u\n", packet->data[SID_byte]);
+  }
+  
+  switch(packet->data[SID_byte]){
+    case 0: 
+      printf("BATT 1 CURRENT: %f\n", *(float *)(packet->data + SID_byte + 1));
+      printf("BATT 1 VOLTAGE: %f\n", *(float *)(packet->data + SID_byte + 5));
+      printf("BATT 1 TEMPRATURE: %f\n", *(float *)(packet->data + SID_byte + 9));
+      break;
+    
+    default:
+      printf("No data found\n");
+  }
+
+  packet->length = (sizeof((char *) packet->data) + 1); /* include the 0 termination */
+  
+  if (packet->length > csp_buffer_data_size()) {
+		return CSP_ERR_NOMEM;
+	}
```

Comment:
Let's do this in the demo_hal file before we try putting data in the buffer to make sure that there's enough data space to avoid a segmentation fault

Reply:
Already updated. However I still need to keep tracing this since we use void pointer here and not sure what kind of effect it will have on data sizes and values such as if I tried to cast it to char * [here](https://github.com/iBacklight/ex2_services/blob/7b0cf18b86a1b69828b88694ab1989b49996e456/Platform/demo/hal/demo_hal.c#L57)?